### ISSUE TITLE
Feature Request: Suggest an edit for note/help

### ISSUE THREAD

**Issue by spencerpogo** [zesterer/ariadne#56]
Would be useful to be able to suggest edits in a note/help message similar to how rust does it. 

Need to be able to provide a new version of the line/file, not sure how that would work.

**Comment by nerdypepper** [comment]
i love the idea of doing this (i'd imagine a lot of effort might go behind the implementation of this). i just wanted to second this as i have a use case for this in `statix` to supply "fix" suggestions. currently, the suggestion is just supplied as part of the message as `Try ... instead`:
![Screenshot_20230108_025831](https://user-images.githubusercontent.com/23743547/212256070-93be7086-67eb-4b34-9631-51877aa0f3ea.png)

**Comment by zesterer** [comment]
Yes, I do a similar thing in [Tao](https://github.com/zesterer/tao). It works for the most part, but it would be nice to have an inline hint.

![image](https://user-images.githubusercontent.com/3268430/212354516-e67301de-ba13-4d8c-a89b-f74b393e8c10.png)

I've been meaning to give this crate the time it needs for a full refactor, but I've yet to find the time (life keeps throwing curveballs at me). I'd definitely want this feature to be part of that.

**Comment by zesterer** [comment]
Closing in favour of #148
